### PR TITLE
Kernelize Start\Fire

### PR DESCRIPTION
**Pull request by Odomontois**
Using basically the same approach as in #681, while parameterizing Start over Fiber into new type class Fibers

### CODE REVIEW

File: kernel/src/main/scala/tofu/Fire.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import scala.util.Either
+import tofu.internal.{EffectComp, Effect2Comp}
+import tofu.internal.carriers.FibersCarrier
+
+trait Fire[F[_]] {
+  def fireAndForget[A](fa: F[A]): F[Unit]
+}
+
+object Fire extends EffectComp[Fire] {
+  final implicit def byCarrier[F[_], Fib[_]](implicit
```
Comment (FunFunFine) [inline kernel/src/main/scala/tofu/Fire.scala:12]: I think it's better to note somehow that it's just for scala 2.12

File: kernel/src/main/scala/tofu/Fire.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+trait Race[F[_]] extends Fire[F] {
+  def race[A, B](fa: F[A], fb: F[B]): F[Either[A, B]]
+  def never[A]: F[A]
+}
+
+object Race extends EffectComp[Race] {
+  def never[F[_], A](implicit race: Race[F]): F[A] = race.never
+}
+
+trait Fibers[F[_], Fib[_]] extends Race[F] {
```
[the remark this answers is not in the document]
Comment (Odomontois) [reply kernel/src/main/scala/tofu/Fire.scala:26]: I tend not to mix type parameter names with existing types. It creates readability hell sometimes

**Review by FunFunFine — DISMISSED** [review]

File: kernel/src/main/scala/tofu/Fire.scala (hunk 1 above)
Comment (Odomontois) [reply kernel/src/main/scala/tofu/Fire.scala:12]: It's not

Comment (Odomontois) [reply kernel/src/main/scala/tofu/Fire.scala:12]: As in #681 this carrier allows `Fib` to be inferred within the macro, just like shapeless whitebox macros do the typelevel calculations we can depend on types that can be unknown if the needed module isn't in the classpath

File: kernel/src/main/scala/tofu/Fire.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+trait Race[F[_]] extends Fire[F] {
+  def race[A, B](fa: F[A], fb: F[B]): F[Either[A, B]]
+  def never[A]: F[A]
+}
+
+object Race extends EffectComp[Race] {
+  def never[F[_], A](implicit race: Race[F]): F[A] = race.never
+}
+
+trait Fibers[F[_], Exit[_], Fib[_]] extends Race[F] {
```
Comment (vilunov) [inline kernel/src/main/scala/tofu/Fire.scala:26]: Why do we need the `Exit` param? Does not seem to be used anywhere

Comment (Odomontois) [reply kernel/src/main/scala/tofu/Fire.scala:26]: - [ ] Actually use Exit

**Review by catostrophe — APPROVED** [review]

**Pull request merged**